crates/network-scheduler/src/scheduler.rs

```diff
@@ -460,3 +474,21 @@ fn add_signature_headers(
         },
     ]);
 }
+
+fn replication_factor(data_size_bytes: u64, num_workers: u64) -> Option<usize> {
+    if data_size_bytes == 0 {
+        return None;
+    }
+    let conf = Config::get();
+    if !conf.dynamic_replication {
+        return Some(conf.replication_factor);
+    }
+    // We aim to fill `dynamic_replication_factor` * total available capacity
```

Comment:
So it's not a replication factor, is it? Maybe something like `used_capacity` would be a better name?

Reply:
OK, will rename to `dynamic_rep_capacity_share`